### PR TITLE
Async chunk loading on paper

### PR DESCRIPTION
**Pull request by masmc05**
This isn't finished or tested, it just shows how i want to implement my idea, it's only on 1.18.2
It already has some in improvements in msec per render (on my 8 cores 16 threads laptop it's like 2-3 times), also it totally disappeared from spark profilers  ([old](https://spark.lucko.me/pKVR2uNUiB) vs [new](https://spark.lucko.me/MAVppG8VKJ))

P.S. if you have a tip how i can make indendations on intellij IDEA like in your project i would be pleased

### CODE REVIEW

**Comment by mikeprimm** [comment]
Cool - I don't use IntellJ (still mostly Eclipse here, for Java anyway).   Main thing is 4 space indentation, versus tabs.

**Comment by mikeprimm** [comment]
Once you're ready, I'll review the change approach, and I might look at porting it back to some earlier versions.  The Forge/Fabric code already avoids main thread loading - the Bukkit API limitations kind of still continue to drive the Spigot/Paper version.

**Comment by FedUpWith-Tech** [comment]
https://stackoverflow.com/questions/41532061/shortcut-for-changing-tabs-spaces-tabs-in-intellij-idea#:~:text=To%20toggle%20between%20tabs%20and,Spaces%20or%20To%20Tabs%20respectively.
Here's the way to change IntelliJ to spaces not tabs @mastermc05

**Comment by masmc05** [comment]
> https://stackoverflow.com/questions/41532061/shortcut-for-changing-tabs-spaces-tabs-in-intellij-idea#:~:text=To%20toggle%20between%20tabs%20and,Spaces%20or%20To%20Tabs%20respectively. Here's the way to change IntelliJ to spaces not tabs @mastermc05

i tried yesterday that, but i think it didn't work because i was amending changes with indentation, but the commit had same tabs. Because i was already sleepy didn't come with a better idea. I'll try to change directly in github the tabs to space, i think it shouldn't be the same problem anymore

**Comment by masmc05** [comment]
image
wow, didn't think i could achieve such performance, it says every second or twice a second about 
like it was like that
image
 like 3-4 times slower, and 4-5 times bigger mspt

**Comment by masmc05** [comment]
[latest.log](https://github.com/webbukkit/dynmap/files/8382592/latest.log)
well, seems like on 16 threads, dynmap is worse than i thought, hopefully in my version it seems fixed

File: DynmapCore/src/main/java/org/dynmap/common/chunk/GenericMapChunkCache.java [hunk 1]
```diff
@@ -840,6 +913,81 @@ else if (!snaparray[i].isEmpty) {
...
+
+		List<DynmapChunk> chunks;
+		if (iterator == null) {
+			iterator = Collections.emptyListIterator();
+			chunks = new ArrayList<>(this.chunks);
+		} else {
+			chunks = new ArrayList<>();
+			iterator.forEachRemaining(chunks::add);
+		}
+//		DynmapCore.setIgnoreChunkLoads(true);
```
[the remark this answers is not in the document]
Comment (mastermc05) [reply DynmapCore/src/main/java/org/dynmap/common/chunk/GenericMapChunkCache.java]: nvm, found a better way

**Comment by masmc05** [comment]
i think is good now, other versions i'll do in another pull request

**Comment by masmc05** [comment]
[before disabled world spawn load](https://github.com/webbukkit/dynmap/files/8389723/2022-03-31-2.log)
[after](https://github.com/webbukkit/dynmap/files/8389724/2022-03-31-3.log)
491k tiles rendered without any issues

**Comment by mikeprimm** [comment]
Looks pretty good - gonna pull and give it a go

**Pull request merged**